pkg/tests/tasks/observability/openshift_monitoring_test.go

```diff
@@ -95,46 +90,99 @@ func TestOpenShiftMonitoring(t *testing.T) {
 		t.LogStep("Deploying Kiali")
 		oc.ApplyTemplate(t, meshNamespace, kialiUserWorkloadMonitoringTmpl, kialiValues)
 
-		t.LogStep("Deploying SMCP")
-		oc.ApplyTemplate(t, meshNamespace, meshTmpl, meshValues)
-		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+		t.NewSubTest("SMCP manageNetworkPolicy false").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, meshNamespace, istiodMonitor)
+				oc.DeleteFromString(t, ns.Foo, istioProxyMonitor)
+				oc.DeleteFromString(t, meshNamespace, enableTrafficMetrics)
+				app.Uninstall(t, app.Httpbin(ns.Foo))
+				oc.DeleteFromTemplate(t, meshNamespace, meshTmpl, meshValues)
+			})
 
-		t.LogStep("Wait until Kiali is ready")
-		oc.WaitCondition(t, meshNamespace, "Kiali", kialiName, "Successful")
+			t.LogStep("Deploying SMCP")
+			oc.ApplyTemplate(t, meshNamespace, meshTmpl, meshValues)
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
 
-		t.LogStep("Verify that Kiali was reconciled by Istio Operator")
-		retry.UntilSuccess(t, func(t test.TestHelper) {
-			output := shell.Executef(t, "oc get kiali %s -n %s -o jsonpath='{.spec.deployment.accessible_namespaces}'", kialiName, meshNamespace)
-			if output != fmt.Sprintf(`["%s"]`, ns.Foo) {
-				t.Errorf(`unexpected accessible namespaces: got '%s', expected: '["%s"]'`, output, ns.Foo)
-			}
+			waitKialiAndVerifyIsReconciled(t)
+
+			t.LogStep("Enable Prometheus telemetry")
+			oc.ApplyString(t, meshNamespace, enableTrafficMetrics)
+
+			t.LogStep("Deploy httpbin")
+			app.InstallAndWaitReady(t, app.Httpbin(ns.Foo))
+			oc.WaitPodsExist(t, ns.Foo)
+			oc.WaitAllPodsReady(t, ns.Foo)
+
+			t.LogStep("Apply Prometheus monitors")
+			oc.ApplyString(t, meshNamespace, istiodMonitor)
+			oc.ApplyString(t, ns.Foo, istioProxyMonitor)
+
+			generateTrafficAndcheckMetrics(t, thanosToken)
 		})
 
-		t.LogStep("Enable Prometheus telemetry")
-		oc.ApplyString(t, meshNamespace, enableTrafficMetrics)
+		t.NewSubTest("SMCP manageNetworkPolicy true").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, meshNamespace, istiodMonitor)
+				oc.DeleteFromString(t, ns.Foo, istioProxyMonitor)
+				oc.DeleteFromString(t, meshNamespace, enableTrafficMetrics)
+				app.Uninstall(t, app.Httpbin(ns.Foo))
+				oc.DeleteFromTemplate(t, meshNamespace, networkPolicy, map[string]string{"namespace": meshNamespace})
+				oc.DeleteFromTemplate(t, meshNamespace, meshTmpl, meshValues)
+			})
 
-		t.LogStep("Deploy httpbin")
-		app.InstallAndWaitReady(t, app.Httpbin(ns.Foo))
-		oc.WaitPodsExist(t, ns.Foo)
-		oc.WaitAllPodsReady(t, ns.Foo)
+			t.LogStep("Deploying SMCP")
+			meshValues["ManageNetworkPolicy"] = "true"
```

Comment:
Aren't templates case sensitive? I guess it shouldn't work...

Reply:
Yes, you are right. Let me do the change and tested again

Reply:
Updated, as you mentioned was not working properly. tested now again and confirmed that is using the network policy

```
--- PASS: TestOpenShiftMonitoring (193.60s)
    --- PASS: TestOpenShiftMonitoring/SMCP_manageNetworkPolicy_false (76.89s)
    --- PASS: TestOpenShiftMonitoring/SMCP_manageNetworkPolicy_true (62.15s)
```